src/io/fasta.rs

```diff
@@ -252,12 +366,73 @@ struct IndexRecord {
 
 
 /// A sequence record returned by the FASTA index.
+#[derive(Debug, PartialEq)]
 pub struct Sequence {
     pub name: String,
     pub len: u64,
 }
 
 
+pub struct IndexedReaderIterator<'a, R: io::Read + io::Seek + 'a> {
+    reader: &'a mut IndexedReader<R>,
+    record: IndexRecord,
+    bases_left: u64,
+    line_offset: u64,
+    buf: Vec<u8>,
+    buf_idx: usize,
+}
+
+
+impl<'a, R: io::Read + io::Seek + 'a> IndexedReaderIterator<'a, R> {
+    fn fill_buffer(&mut self) -> io::Result<()> {
+        assert!(self.bases_left > 0);
+
+        self.buf.clear();
+        let bases_to_read = min(self.buf.capacity() as u64, self.bases_left);
+
+        while self.buf.is_empty() {
```

Comment:
I see, I wasn't even aware that whitespace is allowed in fasta (except newlines). Silly. Can you please add this explanation in a comment above the loop, such that this is not optimized away in the future.